modules/core/04-channel/v2/keeper/packet.go

```diff
@@ -54,21 +54,28 @@ func (k *Keeper) sendPacket(
 		return 0, "", errorsmod.Wrapf(types.ErrInvalidPacket, "constructed packet failed basic validation: %v", err)
 	}
 
+	// Before we do client keeper level checks, we first get underlying base clientID
```

Comment:
What about the check on line 28?

Is there a reason why we wouldn't want the sourceClient to already be resolved before it gets sent to this function?

Reply:
The counterparties in client keeper are stored on the channel IDs

Reply:
Oh right. Yeah that makes sense. 
I think it would be good to add a comment on that, just so it's easy to remember in the future when coming back here :)